src/wr_control_drive_arm/src/ArmControlActionServer.cpp

```diff
@@ -110,17 +113,23 @@ void execute(const control_msgs::FollowJointTrajectoryGoalConstPtr &goal,
             return;
         }
 
+        std::vector<double> velocityCopies{currTargetPosition.velocities};
+
+        for (uint32_t i = 0; i < goal->trajectory.joint_names.size(); ++i) {
+            velocityCopies.at(i) *= abs(namedPositionMonitors.at(goal->trajectory.joint_names.at(i))->getCountsPerRotation());
+        }
+
         const double VELOCITY_MAX = abs(*std::max_element(
-            currTargetPosition.velocities.begin(),
-            currTargetPosition.velocities.end(),
+            velocityCopies.begin(),
+            velocityCopies.end(),
             [](double lhs, double rhs) -> bool { return abs(lhs) < abs(rhs); }));
 
         for (uint32_t i = 0; i < goal->trajectory.joint_names.size(); ++i) {
             auto jointVelocity{JOINT_SAFETY_HOLD_SPEED};
             if (VELOCITY_MAX != 0)
-                jointVelocity = currTargetPosition.velocities.at(i) / VELOCITY_MAX * JOINT_SAFETY_MAX_SPEED;
+                jointVelocity = velocityCopies.at(i) / VELOCITY_MAX * JOINT_SAFETY_MAX_SPEED;
 
-            namedJointMap.at(goal->trajectory.joint_names.at(i))->setTarget(currTargetPosition.positions.at(i), jointVelocity);
+            namedJointMap.at(goal->trajectory.joint_names.at(i))->setTarget(velocityCopies.at(i), jointVelocity);
```

Comment:
The first target is the position target, not the speed, so this should be replaced with what was there before.